### PR TITLE
Merges in updates from upstream.

### PR DESCRIPTION
**Pull request by vpetersson**
@richard-to I just merged in some changes we had made to the fork of OSE in the Pro repo. We used to use this repo as a submodule, but had to make some changes sometime back and ended up quick-fixing them. 

Could you help me sort out this merge? I don't want to lose any of the features that you have made, while I also do not want to lose any changes from upstream.

### CODE REVIEW

**Comment by vpetersson** [comment]
Also, @aljungberg can you take a look at this one too. I looked over the change log upstream and some of these were from you.

File: settings.py [hunk 1]
```diff
@@ -23,29 +23,15 @@
         'default_duration': '10',
         'debug_logging': False,
         'verify_ssl': True,
-        'use_24_hour_clock': False
```
Comment (richard-to) [inline settings.py]: Keep this line. This is a setting for the 24 hour clock

File: utils.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def validate_url(string):
     """
 
     checker = urlparse(string)
-    return bool(checker.scheme in ('rtsp', 'rtmp', 'http', 'https') and checker.netloc)
```
Comment (richard-to) [inline utils.py]: Keep this line. I think you merged in this rtsp and rtmp feature into OSE

File: utils.py [hunk 3]
```diff
@@ -90,12 +90,7 @@ def url_fails(url):
     """
     Accept 200 and 405 as 'OK' statuses for URLs.
     Some hosting providers (like Google App Engine) throws a 405 at `requests`.
-    Can not check RTSP or RTMP, so we just believe it is there (if not OMX Player will terminate, and the next asset is shown -> no ugly error messages)
     """
-    if url.startswith('rtsp://'):
-        return False
-    if url.startswith('rtmp://'):
-        return False
```
Comment (richard-to) [inline utils.py]: Keep this section as well since it checks rtsp and rtmp URIs

File: assets_helper.py [hunk 4]
```diff
@@ -28,7 +30,7 @@ def is_active(asset, at_time=None):
 
     if asset['is_enabled'] and asset['start_date'] and asset['end_date']:
         at = at_time or get_time()
-        return asset['start_date'] <= at <= asset['end_date']
```
Comment (richard-to) [inline assets_helper.py]: Bug fix from GH-202

File: requirements.txt [hunk 5]
```diff
@@ -1,8 +1,7 @@
-#rerun
 Mako==0.7.3
 MarkupSafe==0.15
 PyHAML==0.1.9
-bottle==0.12.07
```
Comment (richard-to) [inline requirements.txt]: Bottle needed to be updated. Can't remember the issue. I think related to rtsp/rtmp pull request?

File: viewer.py [hunk 6]
```diff
@@ -103,17 +112,14 @@ def get_db_mtime(self):
 
 def generate_asset_list():
     logging.info('Generating asset-list...')
-
-    now = datetime.utcnow()
-    enabled_assets = [a for a in assets_helper.read(db_conn) if a['is_enabled']]
-    future_dates = [a[k] for a in enabled_assets for k in ['start_date', 'end_date'] if a[k] > now]
-    deadline = sorted(future_dates)[0] if future_dates else None
```
Comment (richard-to) [inline viewer.py]: Change for: https://github.com/wireload/screenly-ose/commit/5e3828a2f803926fc5cd69ba5dec8b79864ec45b

File: viewer.py [hunk 7]
```diff
@@ -103,17 +112,14 @@ def get_db_mtime(self):
...
+    playlist = assets_helper.get_playlist(db_conn)
+    deadline = sorted([asset['end_date'] for asset in playlist])[0] if len(playlist) > 0 else None
     logging.debug('generate_asset_list deadline: %s', deadline)
 
-    playlist = assets_helper.get_playlist(db_conn)
```
Comment (richard-to) [inline viewer.py]: Also from: https://github.com/wireload/screenly-ose/commit/5e3828a2f803926fc5cd69ba5dec8b79864ec45b

**Pull request closed without merging**